pkg/controller/bundles/bundles_syncer.go

```diff
@@ -3,39 +3,65 @@ package bundles
 import (
 	"context"
 	"fmt"
+	"sync"
 
 	"github.com/go-logr/logr"
 	"github.com/open-cluster-management/leaf-hub-spec-sync/pkg/bundle"
 	"github.com/open-cluster-management/leaf-hub-spec-sync/pkg/controller/helpers"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/client-go/rest"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // LeafHubBundlesSpecSync syncs bundles spec objects.
 type LeafHubBundlesSpecSync struct {
-	log               logr.Logger
-	k8sClient         client.Client
-	bundleUpdatesChan chan *bundle.ObjectsBundle
+	log                  logr.Logger
+	bundleUpdatesChan    chan *bundle.ObjectsBundle
+	k8sClients           []client.Client
+	clientWorkersJobChan chan *clientWorkerJob
+	clientWorkersWG      sync.WaitGroup
+}
+
+// handlerFunc is a clientWorkerJob's handler function.
+type handlerFunc func(context.Context, client.Client, *unstructured.Unstructured)
+
+// clientWorkerJob holds the object than need to be processed and the flag to which defines
+// whether object need to be updated or delete.
+type clientWorkerJob struct {
+	handler handlerFunc
+	obj     *unstructured.Unstructured
 }
 
 // AddLeafHubBundlesSpecSync adds bundles spec syncer to the manager.
-func AddLeafHubBundlesSpecSync(log logr.Logger, mgr ctrl.Manager, bundleUpdatesChan chan *bundle.ObjectsBundle) error {
+func AddLeafHubBundlesSpecSync(log logr.Logger, mgr ctrl.Manager, bundleUpdatesChan chan *bundle.ObjectsBundle,
+	numOfClients int) error {
 	// creates the in-cluster config
 	config, err := rest.InClusterConfig()
 	if err != nil {
 		return fmt.Errorf("failed to get in cluster kubeconfig - %w", err)
 	}
 
-	k8sClient, err := client.New(config, client.Options{})
-	if err != nil {
-		return fmt.Errorf("failed to initialize k8s client - %w", err)
+	// prepare k8s clients
+	k8sClients := make([]client.Client, 0)
```

Comment:
Since the final size of the array is known already (numOfClients) then the array should be initialized with it as a good practice and for higher efficiency in high size cases. Though I realize that in this 1 time run it is negligible.

Feel free to ignore.

Reply:
done